### PR TITLE
feat(node): v14

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#14.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Taking the opportunity of a major version to suggest some breaking changes as well.

The first commit ONLY contains the changes described in the changelog (see above).

The second commit contain some consolidations:
 - Force all encoding parameters to use the `BufferEncoding` type.
 - NodeJS technically normalises all encodings, eg. `hEx` is a valid encoding, however we have the unique opportunity to steer developers in a more consistent direction.
 - Consolidate `Mode` and `Flags` arguments for into types.

I want some opinions on the latter, I understand quite a few people will probably run into issues with this, I'm also worried about dependencies that are not on DT.

### CODE REVIEW

**Comment by SimonSchick** [comment]
@mateus4k I included your change here.

File: types/node/crypto.d.ts
Comment (addaleax) [inline types/node/crypto.d.ts:432]: Find + replace issue?

File: types/node/test/util.ts
Comment (addaleax) [inline types/node/test/util.ts:170]: ditto?

**Review by addaleax — APPROVED** [review]

File: types/node/crypto.d.ts
Comment (SimonSchick) [reply types/node/crypto.d.ts:432]: Good catch, I used a bad regex, I thought I caught all of these bad boys, ty :)

File: types/node-expat/index.d.ts
Comment (stenalpjolly) [inline types/node-expat/index.d.ts:25]: Isn't it
``` encoding: BufferEncoding ```

Comment (SimonSchick) [reply types/node-expat/index.d.ts:25]: Yup, used a regex that was too greedy :)

**Comment by SimonSchick** [comment]
@sandersn can we potentially halt merges going into node typings until this is done?
Last time I had to rebase about 10 times 😭 

I'd also like your thoughts on this, especially the encoding changes since these could have large scale effects.

**Review by Flarna — APPROVED** [review]
Quite some potential to make some people unhappy but for sure an improvment. As it is a new major version it's the only possible timeframe to try this.

**Comment by SimonSchick** [comment]
@flarna any idea what this error means, I fixed the paths in v13...

**Comment by SimonSchick** [comment]
Build failure is, again, unrelated.

**Comment by SimonSchick** [comment]
@sandersn can this get some love? 😛

**Review by ExE-Boss — APPROVED** [review]
The unrelated build failure is caused by `hapi`.

**Review by winston01 — APPROVED** [review]

**Comment by SimonSchick** [comment]
Finally....

**Comment by SimonSchick** [comment]
Jesus christ the bot has issues.

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
Seems Bot doesn't add the `Owner Approved` label anymore which is usually the trigger that PRs get merged soon.
At least the `Other Approved` was added after I approved again.

**Comment by Flarna** [comment]
in a different PR adding a comment helped that bot moved it into the correct state. try the same here...

**Comment by addaleax** [comment]
Fwiw, that PR was https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44034 and I think this PR might need to be rebased to include it (for the node/v13/ directory) anyway?

**Comment by SimonSchick** [comment]
Yea I was hoping that this would be merged before anyone else's node changes come in 😞 
I will just cherry pick the changes for now, a rebase shouldn't be needed yet.

**Review by Flarna — APPROVED** [review]

**Comment by SimonSchick** [comment]
Pinging random DT maintainer: @kieferrm can you look at this?
I'm looking to have both commits merged individually so we can revert the latter if it is __too__ breaking.

**Comment by SimonSchick** [comment]
Can I at least have a comment?
This has been open for 20 days.

**Comment by sandersn** [comment]
Anybody? I'd like to merge this before #44307.

@jkomyno @a-tarasyuk @alvis @r3nya @btoueg @brunoscheufler @smac89 @tellnes @touffy @DeividasBakanas @eyqs @Flarna @Hannes-Magnusson-CK @KSXGitHub @hoo29 @kjin @ajafff @islishude @mwiktorczyk @mohsen1 @n-e @octo-sniffle @galkin @parambirs @eps1lon @SimonSchick @ThomasdenH @WilcoBakker @wwwy3y3 @samuela @kuehlein @j-oliveras @bhongy @chyzwar @trivikr @nguymin4 @yoursunny @qwelias @ExE-Boss @Ryan-Willpower @peterblazejewicz

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
@sanders Why is it needed to approve several times now? Any changes in the approval process?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@SimonSchick thanks for update!
I've only issue with the `fs/promises`, could be due to my local setup, I'll loook into this later (I've been usng fs-extra for async work.
👍

**Comment by sandersn** [comment]
Because of this comment from @SimonSchick?

> Pinging random DT maintainer: @kieferrm can you look at this?

**Comment by sandersn** [comment]
I think I misread that, and actually @SimonSchick wanted a non-squash-merge for this PR.

**Comment by SimonSchick** [comment]
I do anticipate some compatibility issues with the new encoding for packages that are not on DT, I will address those as they come in.

**Comment by LabhanshAgrawal** [comment]
@SimonSchick we're facing trouble updating [`got`](https://github.com/vercel/hyper/pull/4732) and [`spectron`](https://github.com/vercel/hyper/pull/4731) in https://github.com/vercel/hyper due to difference in type of `encoding` between v12 and v14
```
Type 'string' is not assignable to type '"ascii" | "utf8" | "utf-8" | "utf16le" | "ucs2" | "ucs-2" | "base64" | "latin1" | "binary" | "hex" | undefined'
```
And we can't use v14 as we're using electron.

Would it be possible to update v12 types with `BufferEncoding` instead of `string`
I see it being used for `fs` and `child_process` in v12, so I hope it'd be ok to use it in `stream` also.
I can try and send a pr if needed.

**Pull request merged**
